### ISSUE TITLE
%bout hint crashes when in tail position

### ISSUE THREAD

**Issue by Fang-** [urbit/urbit#5690]
All of the following crash:

```hoon
~>  %bout  |-  7
~>  %bout  |^  7  --
~>  %bout  =<($ |.(7))
~>  %bout  (gulf 1 2)
```

The obvious commonality here that they all evaluate `$` on some core. Gates seem to be unaffected though.

```
address 0x2300000288 out of loom!
loom: [0x28000000000 : 0x28080000000)
Assertion '0' failed in noun/events.c:128

bail: oops
bailing out
pier: serf unexpectedly shut down
```

**Comment by joemfb** [comment]
The actual bug is a mis-compilation of dynamic hints in tail position, fixed in #5493. Releasing soon(tm)!

**Comment by joemfb** [comment]
You can workaround it by moving the hints out of tail position: `=+  ~>  %bout  |-  7  -`.
